### PR TITLE
feat: [PLG-2329]: credit card verification widget and billing page changes

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/33-auth-settings/pages/Billing/FooterCreditCard.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        }
+      })
+
+      //save the card details call the save api
+      await saveCard({
+        accountIdentifier: accountId,
+        creditCardIdentifier: res?.setupIntent?.payment_method as string
+      })
+
+      showSuccess('Card is validated and saved')
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/pages/Billing/FooterCreditCard.tsx:70]: Use strings framework

File: src/modules/33-auth-settings/pages/Billing/PaymentMethods.tsx [hunk 2]
```diff
@@ -28,52 +30,78 @@ const cardByImageMap: { [key: string]: string } = {
...
+                    data?.data?.expireYear
+                  }`}
+                </Text>
+              </Layout.Vertical>
+            </Layout.Horizontal>
+          )
+        }
         {/* add card experience to be enabled later */}
         {/* <div className={css.centerText}>
         <Text font={{ variation: FontVariation.BODY }}>{getString('authSettings.billingInfo.addCC')}</Text>
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/pages/Billing/PaymentMethods.tsx:107]: Unused code can be removed

File: src/modules/33-auth-settings/pages/Billing/__tests__/FooterCreditCard.test.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        <FooterCreditCard subscriptionId="" isValid={true} onClose={jest.fn()} clientSecret="" />
+      </TestWrapper>
+    )
+    expect(container).toMatchSnapshot('payment method card snapshot')
+
+    // calling the cancel function
+
+    const cancelButton = getByText('cancel')
+    expect(cancelButton).toBeTruthy()
+    fireEvent.click(cancelButton)
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/pages/Billing/__tests__/FooterCreditCard.test.tsx:49]: Nit: After `fireEvent`, try to add some assertions.

File: src/modules/33-auth-settings/pages/Billing/__tests__/PaymentMethods2.test.tsx
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/pages/Billing/__tests__/PaymentMethods2.test.tsx:1]: Why do we have 2 test files for PaymentMethods?

**Review by mayankVermaHarness — APPROVED** [review]
Added some comments, no blockers though. Please take care of the `string` that isn't using the Strings-framework. Approved.

**Review by tan-nhu — APPROVED** [review]

**Pull request merged**